module-2/backend/week_one.md

```diff
@@ -4,31 +4,66 @@ Fork this respository. Answer the questions to the best of your ability. Try to
+ 
+
 2. What is Sinatra?
+  Sinatra is a web frame work that allows you to make web applications. I think it is simpler than Rails.
 4. What is MVC?
+  Model, View, Controller
 5. Why do we follow conventions when creating our actions/path names in our Sinatra routes?
-6. What types of variables are accessible in our view templates without explicitly passing them?
+  They won't work with the MVC format if we don't.
+6. What types of variables are accessible in our view templates without explicitly passing them? 
+   instance variables  
 7. Given the following block of code, how would I pass an instance variable `count` with a value of `1` to my `index.erb` template?
   
   ```ruby
   get '/horses' do
     erb :index
   end
   ```
+    @count = 1
 
 8. In the same code block, how would I pass a local variable `name` with a value of `Mr. Ed` to the view?
+     @name = "Mr. Ed" 
 9. What's the purpose of ERB?
+  It allows you to run ruby code inside of a HTML document.  
 10. Why do I need a development AND test database?
+  Everytime you run a test, you want to start with the same database and you want to wipe out anything that you added to the database after the test.   
 11. What's responsive design?
+  Responsive design is when you make sure your application behaves the way your want it to when it is viewed on mobile devices or when you change the size of your web browser window.  
 12. What is CRUD and why is it important?
+  CRUD is create, read, update, delete. It is important because it is the base of what you need to make for most web applications and corresponds with the HTTP verbs.  
 13. What does HTTP stand for? 
+  Hypertext Transfer Protocol  
+
 14. What are the two ways to interpolate Ruby in an ERB view template? What's the difference between these two ways?
+  <%= %> and <% %>, The difference is that the one with the equal sign will put the value of some ruby code while the other one will not. It is where you put ruby commands that don't return a value.  
 15. What's an ORM?
-16. What's the most commonly used ORM in ruby (Sinatra & Rails)?
-17. Let's say we have an application with restaurants. There are seven verb + path combinations necessary to provide full CRUD functionality for our restaurant application. List each of the seven combinations, and explain what each is for.
+  object relational mapper  
+
+16. What's the most commonly used ORM in ruby (Sinatra & Rails)? 
+  Active Record  
+17. Let's say we have an application with restaurants. There are seven verb + path combinations necessary to provide full CRUD functionality for our restaurant application. List each of the seven combinations, and explain what each is for.  
+  *GET "/restaurants" - Shows all restaurants  
+  *GET "/restaurants/:id" - Shows one restaurant   
+  *GET "/restaurants/new" - Goes to the form to make a new restaurant  
+  *POST "/restaurants/" - Saves the info from the form and redirects you back to the all restaurants view  

```

Comment:
No trailing slash on the path (`"/restaurants"`), and don't always have to redirect back to the index.